Sources/Sentry/SentrySessionTracker.m

```diff
@@ -72,4 +80,9 @@ - (void)willResignActive {
     self.lastInForeground = [NSDate date];
 }
 
+- (void)willTerminate {
+    NSDate *from = nil == self.lastInForeground ? [NSDate date] : self.lastInForeground;
+    [[SentrySDK currentHub] endSessionWithTimestamp:from];
+}
```

Comment:
`m`: I know there aren't any tests right now for this class and there isn't much logic happening in this method, but still I would prefer having tests for `didBecomeActive`, `willResignActive` and `willTerminate`. Maybe we can split the Sentry logic from the `NSNotificationCenter` logic to make our code testable.

Reply:
Yeah need to dig into how to test those but generally need to test sessions, this class has 0 tests :hides:

Reply:
Session tests were added to pbxproj (got lost when moving to `xcworkspace`: https://github.com/getsentry/sentry-cocoa/pull/508

Still need to test the Tracker though and use the new DateTimeProvider bits